### PR TITLE
Issue #13345: Enable examples tests for MissingJavadocPackageCheck

### PR DESCRIPTION
**Pull request by AmitKumarDeoghoria**
### Title: Enable examples tests for  MissingJavadocPackageCheck
**Related Issue: #13345** 

**Summary**
This pull request aims to enable example tests for the **  MissingJavadocPackageCheck** in the project. The implementation focuses on ensuring that   MissingJavadocPackage are correctly validated through comprehensive test cases, improving overall code reliability and quality.

**Key Changes Made**

- Conversion of txt file to Java File
- Enable the test cases by adding expected violations in the expected array for all example tests
- Made corresponding changes in the xml and xml.template file

### CODE REVIEW

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocpackage/Example2.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="MissingJavadocPackage"/>
+  </module>
+</module>
+*/
+// xdoc section -- start
+/*
```
Comment (AmitKumarDeoghoria) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocpackage/Example2.java:9]: violation is not raised for the above block comment for some reason. @romani can you please provide some insights.

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocpackage/Example2.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * Block comment is not a javadoc
+ */
```
Comment (AmitKumarDeoghoria) [inline src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocpackage/Example2.java:11]: violation is not raised for the above block comment. @romani can you please help and provide some insight.

**Comment by romani** [comment]
GitHub, generate website

File: src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocpackage/Example2.java (hunk 2 above)
Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocpackage/Example2.java:11]: https://checkstyle.org/checks/javadoc/missingjavadocpackage.html#Description

>Checks for missing package definition Javadoc comments in package-info.java files.

Please rename file.

Comment (romani) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocpackage/Example2.java:11]: All files should be named `package-info.java` and located in different packages/folders.

There should be no `class` in such files, so please add such files to suppressions

Comment (AmitKumarDeoghoria) [reply src/xdocs-examples/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocpackage/Example2.java]: Done

**Comment by AmitKumarDeoghoria** [comment]
GitHub, generate website

**Comment by AmitKumarDeoghoria** [comment]
CI is failing due to same <P id=package-info-code> since both files have same name it is causing error in CI. What should we do to rectify that @romani

**Comment by romani** [comment]
```
 ~/java/github/romani/checkstyle [AmitKumarDeoghoria/misjdoc L|✚ 3⚑ 1] 
 $ git diff
diff --git a/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java b/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java
index 2e36e91..ad0f609 100644
--- a/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java
@@ -2119,10 +2119,17 @@ public class XdocsPagesTest {
         final String id = idAttribute.getTextContent();
         final String expectedId = String.format(Locale.ROOT, "%s-%s", exampleName,
                 exampleType);
-        assertWithMessage(fileName
+        if (expectedId.startsWith("package-info")) {
+            assertWithMessage(fileName
+                + ": paragraph before example macro should have the expected id value")
+                .that(id)
+                .endsWith(expectedId);
+        } else {
+            assertWithMessage(fileName
                 + ": paragraph before example macro should have the expected id value")
                 .that(id)
                 .isEqualTo(expectedId);
+        }
     }
 
     private static Node getPrecedingParagraph(Node macro) {
diff --git a/src/xdocs/checks/javadoc/missingjavadocpackage.xml b/src/xdocs/checks/javadoc/missingjavadocpackage.xml
index b0d7879..7873171 100644
--- a/src/xdocs/checks/javadoc/missingjavadocpackage.xml
+++ b/src/xdocs/checks/javadoc/missingjavadocpackage.xml
@@ -27,7 +27,7 @@
       </subsection>
 
       <subsection name="Examples" id="Examples">
-        <p id="package-info-config">
+        <p id="javadoc-package-info-config">
             To configure the check:
         </p>
         <source>
@@ -37,15 +37,14 @@
   &lt;/module&gt;
 &lt;/module&gt;
         </source>
-        <p id="package-info-code">Example1:</p>
+        <p id="javadoc-package-info-code">Example1 of package-info.java:</p>
         <source>
-/*
- * Block comment is not a javadoc
+/**
+ * Provides API classes
  */
-package com.puppycrawl.tools.checkstyle.checks.javadoc.missingjavadocpackage.nojavadoc;
-// violation above 'Missing javadoc for package-info.java file'
+package com.puppycrawl.tools.checkstyle.checks.javadoc.missingjavadocpackage.javadoc;
         </source>
-        <p id="package-info-code">Example2:</p>
+        <p id="nojavadoc-package-info-code">Example2 of package-info.java:</p>
         <source>
 /*
  * Block comment is not a javadoc
diff --git a/src/xdocs/checks/javadoc/missingjavadocpackage.xml.template b/src/xdocs/checks/javadoc/missingjavadocpackage.xml.template
index 2b62505..115300a 100644
--- a/src/xdocs/checks/javadoc/missingjavadocpackage.xml.template
+++ b/src/xdocs/checks/javadoc/missingjavadocpackage.xml.template
@@ -27,7 +27,7 @@
       </subsection>
 
       <subsection name="Examples" id="Examples">
-        <p id="package-info-config">
+        <p id="javadoc-package-info-config">
             To configure the check:
         </p>
         <macro name="example">
@@ -35,13 +35,13 @@
                  value="resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocpackage/javadoc/package-info.java"/>
           <param name="type" value="config"/>
         </macro>
-        <p id="package-info-code">Example1:</p>
+        <p id="javadoc-package-info-code">Example1 of package-info.java:</p>
         <macro name="example">
           <param name="path"
-                 value="resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocpackage/nojavadoc/package-info.java"/>
+                 value="resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocpackage/javadoc/package-info.java"/>
           <param name="type" value="code"/>
         </macro>
-        <p id="package-info-code">Example2:</p>
+        <p id="nojavadoc-package-info-code">Example2 of package-info.java:</p>
         <macro name="example">
           <param name="path"
                  value="resources/com/puppycrawl/tools/checkstyle/checks/javadoc/missingjavadocpackage/nojavadoc/package-info.java"/>
✔ ~/java/github/romani/checkstyle [AmitKumarDeoghoria/misjdoc L|✚ 3⚑ 1] 
```

**Comment by romani** [comment]
GitHub, generate website

**Review by romani — APPROVED** [review]

**Pull request merged**